hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/InProcessHapiTestNode.java

```diff
@@ -90,21 +151,123 @@ public void waitForActive(long seconds) {
                         "node " + nodeId + ": Interrupted while sleeping in waitForActive busy loop", e);
             }
         }
+
+        // We timed out.
+        throw new TimeoutException(
+                "node " + nodeId + ": Waited " + seconds + " seconds, but node did not become active!");
     }
 
     @Override
-    public void stop() {
-        if (th != null) {
+    public void shutdown() {
+        if (th != null && (th.hedera.isFrozen() || th.hedera.isActive())) {
             if (th.hedera != null) th.hedera.shutdown();
             th.interrupt();
-            th = null;
+
+            // This is a hack, but it's the best I can do without classloader isolation and without a systematic
+            // way to shut down a node. Normally, nodes are shutdown by existing the JVM. However, we don't want to
+            // do that here, because we're in-process! So what I'm going to do is:
+            //  1. Search through all the threads in the JVM to find all of them with a callstack with "WorkerThread"
+            //     in the stack. This means they were called by the WorkerThread. Only objects that are part of the
+            //     node will be in this callstack.
+            //  2. For each such thread, stop it. Thread.stop is deprecated and discouraged, because it is almost
```

Comment:
Note, we **should** be able to call `Thread.interrupt()` on the thread and get a clean shutdown.  If we do not, then the platform code is incorrect and we need to fix the bugs preventing a proper shutdown in response to interrupt.

Reply:
I agree, that would be preferable. Really I want to add some API to platform to "shutdown" and let it handle all the details.